src/vs/platform/environment/electron-main/environmentMainService.ts

```diff
@@ -68,6 +69,9 @@ export class EnvironmentMainService extends NativeEnvironmentService implements
 	@memoize
 	get useCodeCache(): boolean { return !!this.codeCachePath; }
 
+	@memoize
+	override get userRoamingDataHome(): URI { return this.appSettingsHome; }
```

Comment:
@sandy081 its a bit hard for me to understand the consequences. Is it that this would essentially put profiles under the `file` scheme here:

https://github.com/microsoft/vscode/blob/0e638c988a64a9d68d78027621ccfbe5ca680977/src/vs/platform/userDataProfile/common/userDataProfile.ts#L218

And thus help for the global storage path here:

https://github.com/microsoft/vscode/blob/0e638c988a64a9d68d78027621ccfbe5ca680977/src/vs/platform/userDataProfile/common/userDataProfile.ts#L45

But would it have side effects other than that? Is there a file system provider in `electron-main` for `vscode-userdata` that behaves different than `file` scheme? Or do we have code that probes on file scheme to make certain assumptions now?